### DISCUSSION TITLE
Add support for ReScript language

### DISCUSSION THREAD

**Discussion by sprkv5** [boyter/scc#398] [Ideas]
ReScript is a new JS-like language derived from ReasonML and OCaml that compiles to JavaScript.

Since scc already supports ReasonML, adding support for ReScript would be a similar effort. Kindly look into this.

**Comment by boyter** [comment]
Do you happen to have a link to the language spec? What file extensions does it use? If I can see that its very easy to add.

**Reply by sprkv5** [reply]
Just like [ReasonML's overview page](https://reasonml.github.io/docs/en/overview), there's [ReScript's overview page](https://rescript-lang.org/docs/manual/latest/overview), that provides a high level overview of the syntax. Kindly let me know if this won't suffice.

As for the extensions, there are [two](https://rescript-lang.org/docs/gentype/latest/usage#interface-resi-and-implementation-res-files): Interface (**.resi**) and Implementation (**.res**) files.

**Reply by DZakh** [reply]
Looks like this is what you need https://rescript-lang.org/docs/manual/latest/overview

I've copy-pasted the ReasonML config and done some changes. I hope it'll help:
```json
"ReScript": {
  "complexitychecks": [
    "for ",
    "for(", // It's always formatted using built-in formatter with a space, so is this really needed?
    "if ",
    "if(",
    "switch ",
    "while ",
    "else ", // else if ???
    "|| ",
    "&& ",
    "!= ",
    "!== ",
    "== ",
    "=== "
  ],
  "extensions": [
    "res",
    "resi"
  ],
  "line_comment": [
    "//"
  ],
  "multi_line": [
    [
      "/*",
      "*/"
    ]
  ],
  "quotes": [
    {
      "end": "`",
      "start": "`"
    },
    {
      "end": "\"",
      "start": "\""
    }
  ]
}
```

**Reply by DZakh** [reply]
There are also `'` quotes for chars, but probably they shouldn't be listed in the config

**Reply by DZakh** [reply]
Also, there's a [ternary operator](https://rescript-lang.org/docs/manual/latest/overview#if-else) using `?` and `:`. I wonder why it's not included for both `ReasonMl` and `JavaScript` languages.
